engine/runtime/src/main/java/org/enso/interpreter/runtime/error/DataflowError.java

```diff
@@ -53,11 +54,14 @@ public final class DataflowError extends AbstractTruffleException implements Ens
    */
   public static DataflowError withDefaultTrace(State state, Object payload, Node location) {
     assert payload != null;
+    var ensoCtx = EnsoContext.get(location);
+    var dataflowStacktraceCtx = ensoCtx.getBuiltins().context().getDataflowStackTrace();
+    // dataflowStacktraceCtx is a compiler constant, because it is a constructor of a builtin
+    // type.
+    CompilerAsserts.partialEvaluationConstant(dataflowStacktraceCtx);
```

Comment:
This helps with the performance. The compiler is most probably not able to determine that `dataflowStacktraceCtx` is actually a constant (reference to a constructor of a builtin type).

Reply:
> This helps with the performance.

That would be surprising. It is a `CompilerAsserts` - e.g. not a directive, but only an assert that something is true. E.g. either `dataflowStacktraceCtx` is constant or the compilation is going to fail. The fact whether it is constant or not isn't changed by the assert. The fact whether compilation goes or not is influenced by the assert.